components/invitation_modal/invitation_modal_guests_step.jsx

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {isEmail} from 'mattermost-redux/utils/helpers';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import InviteIcon from 'components/svg/invite_icon';
+import CloseCircleIcon from 'components/svg/close_circle_icon';
+
+import ChannelsInput from 'components/widgets/inputs/channels_input.jsx';
+import UsersEmailsInput from 'components/widgets/inputs/users_emails_input.jsx';
+
+import BackIcon from 'components/svg/back_icon';
+
+import './invitation_modal_guests_step.scss';
+
+import debouncePromise from 'utils/debounce_promise.jsx';
+import {t} from 'utils/i18n.jsx';
+
+export default class InvitationModalGuestsStep extends React.Component {
+    static propTypes = {
+        goBack: PropTypes.func,
+        myInvitableChannels: PropTypes.array.isRequired,
+        currentTeamId: PropTypes.string.isRequired,
+        searchProfiles: PropTypes.func.isRequired,
+        onEdit: PropTypes.func.isRequired,
+        onSubmit: PropTypes.func.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            customMessageOpen: false,
+            customMessage: '',
+            usersAndEmails: [],
+            channels: [],
+        };
+    }
+
+    onUsersEmailsChange = (usersAndEmails) => {
+        this.setState({usersAndEmails});
+        this.props.onEdit();
+    }
+
+    onChannelsChange = (channels) => {
+        this.setState({channels});
+        this.props.onEdit();
+    }
+
+    onMessageChange = (e) => {
+        this.setState({customMessage: e.target.value});
+        this.props.onEdit();
+    }
+
+    usersLoader = async (term) => {
+        if (isEmail(term)) {
+            return [];
+        }
+        const {data} = await this.props.searchProfiles(term);
+        return data;
+    }
+    debouncedUsersLoader = debouncePromise(this.usersLoader, 150);
+
+    channelsLoader = async (value) => {
```

Comment:
```suggestion
    channelsLoader = (value) => {
```

Reply:
In this special case, I use the async to generate a function that returns a promise, which is the expected value for the `channelsLoader`, I can remove the async, but anyway, the component is going to expect a promise.